astropy/coordinates/sky_coordinate.py

```diff
@@ -41,6 +42,37 @@ def FRAME_ATTR_NAMES_SET():
     return out
 
 
+class SkyCoordDataInfo(BaseInfo):
+    attrs_from_parent = set(['unit'])
+
+    @staticmethod
+    def default_format(val):
```

Comment:
OK, I see how this would get messy! I think it is fine to leave it for future optimization, since nothing you do here prevents us from turning this into a string or a different function later.